src/queue/mapping.rs

```diff
@@ -363,10 +371,15 @@ impl Mapping {
             if output.len() >= std::mem::size_of::<u32>() {
                 let topic_len = u32::from_le_bytes(output[..4].try_into().unwrap()) as usize;
                 if output.len() >= 4 + topic_len {
-                    // extract topic and msg using the topic length prefix
-                    let topic = String::from_utf8_lossy(&output[4..4 + topic_len]).to_string();
-                    let msg = output[4 + topic_len..].to_vec();
-                    data[data_idx] = (topic, msg);
+                    if copy_topic {
+                        // extract topic and msg using the topic length prefix
+                        let topic_str = std::str::from_utf8(&output[4..4 + topic_len])
```

Comment:
Ditto, std::str::from_utf8 faster than String::from_utf8_lossy?

Reply:
i'm not sure tbh, i think `String::from_utf8_lossy` would be similar as [docs say](https://doc.rust-lang.org/std/string/struct.String.html#method.from_utf8_lossy):

```
This function returns a [Cow<'a, str>](https://doc.rust-lang.org/std/borrow/enum.Cow.html). If our byte slice is invalid UTF-8, then we need to insert the replacement characters, which will change the size of the string, and hence, require a String. But if it’s already valid UTF-8, we don’t need a new allocation. This return type allows us to handle both cases.
```